### PR TITLE
DDF-1320: Removing dependency on GraphViz for JavaDoc, fixing JavaDoc Profile

### PR DESCRIPTION
**Pull request by andrewkfiedler**
- Switched from umlgraph to doclet.
- Changed output directory to be under project-information/api.
- Updated javadoc profile to skip checkstyle.

[Review on Reviewable](https://reviewable.io/reviews/codice/ddf-parent/30)

### CODE REVIEW

**Comment by andrewkfiedler** [comment]
@garrettfreibott can you hero?

**Comment by parret** [comment]
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-parent/30)

File: pom.xml [hunk 1]
```diff
@@ -431,8 +431,8 @@
                         <doclet>org.umlgraph.doclet.UmlGraphDoc</doclet>
                         <docletArtifact>
                             <groupId>org.umlgraph</groupId>
-                            <artifactId>umlgraph</artifactId>
-                            <version>5.6.6</version>
+                            <artifactId>doclet</artifactId>
```
Comment (lessarderic) [inline pom.xml]: Can you remove the GraphViz comment above (lines 412-415)?

File: pom.xml [hunk 2]
```diff
@@ -947,8 +946,8 @@
                                 </goals>
                                 <phase>pre-site</phase>
                                 <configuration>
-                                    <outputDirectory>${project.build.directory}/docs/</outputDirectory>
-                                    <reportOutputDirectory>${project.reporting.outputDirectory}/docs/
+                                    <outputDirectory>${project.build.directory}/project-info/</outputDirectory>
```
Comment (lessarderic) [inline pom.xml]: We should use a property instead of hard-coding project-info.

**Comment by parret** [comment]
Fully built and passed tests with this PR and the one in ddf-support :thumbsup: 

---

Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-parent/30)

**Comment by mjmeno** [comment]
is there a sample we can look at?  I'm willing to just look over @andrew-fiedler's shoulder.

**Comment by lessarderic** [comment]
@andrew-fiedler, @garrettfreibott, don't forget to address my comments before merging :-)

**Comment by mjmeno** [comment]
You can make this a separate ticket, but if it's easy please update the license.  Right now it says All Rights Reserved but instead it should be Creative Commons w/ Attribution (ask @ricklarsen how the docs are licensed).

**Comment by ricklarsen** [comment]
"This work is licensed under a Creative Commons Attribution 4.0 International License (http://creativecommons.org/licenses/by/4.0/)"

**Comment by andrewkfiedler** [comment]
:+1:

**Comment by parret** [comment]
:+1:

**Comment by parret** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

<sup>**[pom.xml, line 434 [r1]](https://reviewable.io:443/reviews/codice/ddf-parent/30#-JtdeaF-k4kHLn2r7rsl-r1-434)** ([raw file](https://github.com/codice/ddf-parent/blob/297a9eb87a1b282cd387c8b1c36f1f00031d293f/pom.xml#L434)):</sup>
Removed the graphviz comment.

---

<sup>**[pom.xml, line 949 [r1]](https://reviewable.io:443/reviews/codice/ddf-parent/30#-JtdeaF-k4kHLn2r7rsl-r1-949)** ([raw file](https://github.com/codice/ddf-parent/blob/297a9eb87a1b282cd387c8b1c36f1f00031d293f/pom.xml#L949)):</sup>
Updated to use a property.

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-parent/30)

**Comment by parret** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-parent/30)

**Pull request merged**